examples/src/java/org/pantsbuild/example/README.md

```diff
@@ -211,6 +211,57 @@ Toolchain
 Pants uses [Ivy](http://ant.apache.org/ivy/) to resolve `jar` dependencies. To change how Pants
 resolves these, configure `resolve.ivy`.
 
+Starting release `1.4.0.dev26`, Pants added an option to pick [coursier](https://github.com/coursier/coursier)
+as the JVM 3rdparty resolver, with performance improvement being the main motivation. The goal is to retire ivy
+eventually. Example config to use coursier:
+
+    :::ini
+    # This will turn on coursier and turn off ivy.
+    [resolver]
+    resolver: coursier
+
+    [resolve.coursier]
+    # jvm option in case of large resolves
+    jvm_options: ['-Xmx4g', '-XX:MaxMetaspaceSize=256m']
+    # The opposite of resolve.ivy.soft_excludes
+    allow_global_excludes: False
+
+    [cache.resolve.coursier]
+    # In order to enable remote caching, need to relativize the classpath entries serialized.
+    # https://github.com/pantsbuild/pants/issues/5187
+    # We still want to compute fingerprints so we can validate or invalidate,
+    # but do not want to check local or remote cache.
+    read: False
+    write: False
+
+    [export]
+    # Same if needed for large resolves
+    jvm_options: ['-Xmx4g', '-XX:MaxMetaspaceSize=256m']
+
+    [coursier]
+    # Change the following if you choose to [build coursier jar from scratch](https://github.com/coursier/coursier/blob/master/DEVELOPMENT.md#build-with-pants)
+    # or to fetch from different location.
+    bootstrap_jar_url: <url>
+    version: <version>
+
+    fetch_options: [
+        # Specify maven repos
+        '-r', 'https://repo1.maven.org/maven2/',
```

Comment:
Are there examples for custom repo URL formats using placeholders? The equivalent of this in an ivysettings:
```
<url name="snowchain/binhost">
        <artifact pattern="https://foo.com/bar/[organisation]/[artifact]/[revision]/[artifact]-[revision].[ext]" />
      </url>
```

Reply:
For example, I just tried to emulate that with this:

`'-r', 'ivy:https://foo.com/bar/[organisation]/[artifact]/[revision]/[artifact]-[revision].[ext]'`

But unfortunately it replaces the `[artifact]` placeholder with the string "ivy", instead of with the name of my artifact. So maybe ivy emulation is broken?

Reply:
Turns out that [artifact] should be replaced with [module]. So this is an example of the kind of thing that should go into a "how to migrate from ivy" tutorial!!

Reply:
> But unfortunately it replaces the [artifact] placeholder with the string "ivy", instead of with the name of my artifact. So maybe ivy emulation is broken?

if you suspect resolving against ivy repo is broken, please file a ticket at https://github.com/coursier/coursier with a minimal repro.

Reply:
Ugh but there are still other issues. For example, it appears to insist on the presence of an ivy.xml file, which Ivy itself does not.

Reply:
OK, the behavior is weird. It *assumes* that your pattern ends with `/[artifact].[ext]` so that it can substitute artifact=ivy, ext=xml and fetch an ivy.xml. But if your pattern doesn't end with those, it'll still do the substitution and try to fetch the file . So, e.g., if your pattern ends with `/[module].jar` it'll fetch the jar and then try to parse it as XML... 

Ivy deals with this by having two separate settings - one for the artifacts and one for the ivy.xml.